### DISCUSSION TITLE
Como saber se uma transação é real ou falsa pelo ID?

### DISCUSSION THREAD

**Discussion by gabrielribeiro** [bacen/pix-api#615] [Q&A] [answered]
Gostaria de automatizar a verificação de autenticidade de uma transação pix para verificar se é real ou falsa. Existe alguma API para isso? Obrigado desde já e parabéns pelo ótimo trabalho!

**Comment by rubenskuhl** [comment]
Não, e isso com certeza violaria sigilo bancário, além de permitir ataques de enumeração.

**Comment by leolima77** [comment]
Não sei se entendi sua questão mas posso afirmar, se a transação chegou na sua infra via RSFN, é verdadeira.

**Reply by gabrielribeiro** [reply]
Exemplo: recebo comprovantes via pix o dia todo mas não consigo verificar um por um (manualmente) se o dinheiro caiu na conta de destino, então gostaria de automatizar pra saber se algum comprovante é fake.

**Reply by leolima77** [reply, marked as answer]
por comprovante (imagem) vai ser difícil, vc deve se perguntar se precisa verificar esses comprovantes. o dinheiro que vc mencionou, é cash-in de um qrcode? se for vc pode fazer a relação do pagamento com o txId e o cpf do pagador (mas com o txId já funcionaria).

se o recebimento é oriundo de uma transferencia, vc pode checar o cpf do pagador (se vc tiver o dado do pagador previamente) mas tbm é complicado

mas se o seu volume de cash-in for muito grande, nao faz sentido ficar verificando comprovantes, vc deve atrela-los sistemicamente com a ordem de pagto (qrcode)

**Reply by gabrielribeiro** [reply]
Tenho o CPF do pagador! Não é QR Code, é pagamento por chave pix (desculpe usar os termos simples, não conheço os detalhes do projeto pix).
O que seria o txId? É aquele número no comprovante que o banco gera?

**Reply by leolima77** [reply]
txId é o identificador da ordem de pagto (qrcode dinâmico e estático). o que vem no comprovante do banco que vc mencionou deve ser o e2e (end to end id) que é o identificador da transação no BC. 

é preciso entender a sua operaçao para automatizar o fluxo.

**Reply by rubenskuhl** [reply]
> Tenho o CPF do pagador! Não é QR Code, é pagamento por chave pix (desculpe usar os termos simples, não conheço os detalhes do projeto pix). O que seria o txId? É aquele número no comprovante que o banco gera?

Algumas instituições disponibilizam extratos via API ou arquivo... se você não precisa de uma resposta imediata, pode ser uma opção.

**Reply by gabrielribeiro** [reply]
Obrigado, senhores!
